File: task_equivalents/llm_translations/llama/base/r/data_manipulation_preprocessing/01_impute_encode_scale.R
library(readr)
library(dplyr)

Auto <- read_csv('data/csv/Auto.csv', na = "?", 
                 col_types = cols(name = col_character())) %>%
  drop_na()

Auto$mpg01 <- ifelse(Auto$mpg > median(Auto$mpg, na.rm = TRUE), 1, 0)
Auto <- Auto %>% select(-mpg, -name)

var_no_scale <- c('cylinders', 'year', 'origin', 'mpg01')
var_to_scale <- c('displacement', 'horsepower', 'weight', 'acceleration')

scaled_var <- scale(Auto[, var_to_scale])
df <- cbind.data.frame(scaled_var, Auto[, var_no_scale])

x <- df %>% select(-mpg01)
y <- df$mpg01

File: task_equivalents/llm_translations/llama/base/python/data_manipulation_preprocessing/01_impute_encode_scale.py
import pandas as pd
from sklearn.preprocessing import StandardScaler

# Load data
auto = pd.read_csv('data/csv/Auto.csv', na_values='?').set_index('Unnamed: 0' if 'Unnamed: 0' in pd.read_csv('data/csv/Auto.csv', nrows=0).columns else None)
auto = auto.dropna()

# Create a new column 'mpg01'
auto['mpg01'] = (auto['mpg'] > auto['mpg'].median()).astype(int)

# Drop unnecessary columns
auto = auto.drop(['mpg', 'name'], axis=1)

# Scale continuous variables; keep binary/categorical-coded columns unscaled.
var_no_scale = ['cylinders', 'year', 'origin', 'mpg01']
var_to_scale = ['displacement', 'horsepower', 'weight', 'acceleration']

scaler = StandardScaler()
scaled_var = scaler.fit_transform(auto[var_to_scale])
scaled_df = pd.DataFrame(scaled_var, columns=var_to_scale, index=auto.index)

df = pd.concat([scaled_df, auto[var_no_scale]], axis=1)

x = df.iloc[:, :-1]
y = df['mpg01']